client/src/pages/GroupExpenses/Components/GroupExpenseTable.tsx

```diff
@@ -0,0 +1,195 @@
+import * as React from 'react';
+import { Checkbox, Stack, Typography } from '@mui/material';
+import { GridColDef, GridRenderCellParams } from '@mui/x-data-grid';
+
+import {
+  splitDataWithId,
+  GroupExpenseTableProps,
+  GroupExpenseAsPaid,
+  GroupExpenseData,
+  GroupExpenseDataWithId,
+} from '../../../interfaces/interfaces';
+import { BackgroundBox, ExpensesDataGrid } from './styles';
+import { isoToFormattedDate } from '../../../utils/utils';
+import { categories } from '../../../constants/constants';
+import { theme } from '../../../assets/styles';
+/**
+ * Expenses Group table component
+ *
+ * @param {ExpenseData[]} expenses - expenses to display
+ * @returns {JSX.Element} - Group expenses table
+ */
+export const GroupExpensesTable: React.FC<GroupExpenseTableProps> = ({
+  groupExpenses,
+  handleMarkAsPaidGroupExpense,
+  currUserId,
+}) => {
+  const [newGroupExpenses, setNewGroupExpenses] = React.useState<
+    GroupExpenseDataWithId[]
+  >([]);
+
+  const splitWithId = (data: GroupExpenseData) => {
+    const newSplit: splitDataWithId[] = [];
+
+    data.split.forEach(
+      (splitData, index) =>
+        (newSplit[index] = {
+          id: data.id,
+          userId: splitData.userId,
+          username: splitData.username,
+          hasPaid: splitData.hasPaid,
+          shareAmount: splitData.shareAmount,
+        }),
+    );
+
+    return newSplit;
+  };
+
+  React.useEffect(() => {
+    const updatedGroupExpense: GroupExpenseDataWithId[] = [];
+    groupExpenses.forEach(
+      (data, index) =>
+        (updatedGroupExpense[index] = {
+          id: data.id,
+          title: data.title,
+          totalAmount: data.totalAmount,
+          category: data.category,
+          createdAt: data.createdAt,
+          split: splitWithId(data),
+        }),
+    );
+    setNewGroupExpenses(updatedGroupExpense);
+  }, [groupExpenses]);
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Why do we disable this here?

Reply:
Because the linter doesn't like the any there, but that is what i gotta use for renderCell, I tried specifying but it got angry at me

Reply:
Ok SGTM